rust/src/ketama.rs

```diff
@@ -40,13 +42,28 @@ impl KetamaPool {
         };
         self.ranking[ranking_idx % self.ranking.len()].index as usize
     }
+
+    /// Adds the appropriate rankings for the new node
+    pub fn add_node(&mut self, server_name: &str) {
+        create_server_ranking(&[server_name], &mut self.ranking);
+    }
+
+    /// Removes the rankings corresponding to the server name
+    pub fn remove_node(&mut self, server_name: &str) {
+        self.ranking
+            .retain(|ranking| ranking.server_name == server_name)
+    }
+
+    pub fn get_node(&self, key: &str) -> &str {
+        let slot = self.get_slot(key);
+        self.ranking[slot].server_name.as_ref()
+    }
```

Comment:
this function is undocumented, and it is completely broken per my other comment.
even though a test for it exists, it is frankly not a good test.